src/azul/queues.py

```diff
@@ -377,19 +380,24 @@ def manage_lambdas(self, queues: Mapping[str, Queue], enable: bool):
             def submit(f, *args, **kwargs):
                 futures.append(tpe.submit(f, *args, **kwargs))
 
+            indexer = load_app_module('indexer')
+            queues_to_lambda_names = {
+                lambda_.queue: [config.indexer_function_name(lambda_.lambda_name)]
+                for lambda_ in indexer.app.handler_map.values()
+                if hasattr(lambda_, 'queue')
+            }
+            # Since the indexer lambda writes to the notifications queue, we
+            # must deactivate it also.
+            notifications_queue = config.notifications_queue_name()
+            queues_to_lambda_names[notifications_queue].append(config.indexer_name)
             for queue_name, queue in queues.items():
-                if queue_name == config.notifications_queue_name():
-                    submit(self._manage_lambda, config.indexer_name, enable)
-                    submit(self._manage_sqs_push, config.indexer_name + '-contribute', queue, enable)
-                elif queue_name == config.notifications_queue_name(retry=True):
-                    submit(self._manage_sqs_push, config.indexer_name + '-contribute_retry', queue, enable)
-                elif queue_name == config.tallies_queue_name():
-                    submit(self._manage_sqs_push, config.indexer_name + '-aggregate', queue, enable)
-                elif queue_name == config.tallies_queue_name(retry=True):
-                    # FIXME: Brittle coupling between the string literal below and
-                    #        the handler function name in app.py
-                    #        https://github.com/DataBiosphere/azul/issues/1848
-                    submit(self._manage_sqs_push, config.indexer_name + '-aggregate_retry', queue, enable)
+                try:
+                    lambda_names = queues_to_lambda_names[queue_name]
+                except KeyError:
+                    assert 'fail' in queue_name
```

Comment:
The `"-fail`" suffix is encapsulated in `azul.config` and allowing the substring to occur anywhere in the string, not anchored at boundaries, might lead to false positives depending on the substring. Can we make this both stricter and less brittle?